homeassistant/helpers/config_entry_oauth2_flow.py

```diff
@@ -257,6 +257,11 @@ async def async_step_creation(
     ) -> Dict[str, Any]:
         """Create config entry from external data."""
         token = await self.flow_impl.async_resolve_external_data(self.external_data)
+        # Force float for non-compliant oauth2 providers
+        try:
+            token["expires_in"] = float(token["expires_in"])
+        except ValueError:
+            pass
```

Comment:
We should not fail silently. We should log an error that we can't convert and then abort here or else you'll blow up on the next line!